### PR TITLE
README: Change documentation from python to golang.

### PR DESCRIPTION
**Pull request by shettyg**
Signed-off-by: Gurucharan Shetty <guru@ovn.org>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,35 @@
-# How to Use Open Virtual Networking With Kubernetes
+# How to use Open Virtual Networking With Kubernetes
```
Comment (tong101) [inline README.md]: s/With/with

File: README.md [hunk 2]
```diff
@@ -1,38 +1,35 @@
...
-
-In the "overlay" mode, OVN can create a logical network amongst containers
-running on multiple hosts.  In this mode, OVN programs the Open vSwitch
-instances running inside your VMs or directly on bare metal hosts.
-
-For both the modes to work, a user has to install and start Open vSwitch in
-each VM/host that he plans to run his containers.
+OVN provides network virtualization to containers.  In the "overlay" mode,
+OVN can create a logical network amongst containers running on multiple hosts.
+In this mode, OVN programs the Open vSwitch instances running inside your VMs
```
Comment (tong101) [inline README.md]: To avoid confusion with tenant VMs, can you change it to host VMs?

File: README.md [hunk 3]
```diff
@@ -1,38 +1,35 @@
...
+or directly on bare metal hosts.
 
 Installing Open vSwitch is out of scope of this documentation.  You can read
 [INSTALL.rst] for that.  That documentation inturn links to platform specific
 installations.  If you use packages to install OVS, you should install both
 OVS and OVN related packages.  You can also read the following quick-start
-quide for Ubuntu that installs OVS and OVN from source:  [INSTALL.UBUNTU.md]
+quide for Ubuntu that installs OVS and OVN from source and packages:
```
Comment (tong101) [inline README.md]: s/quide/guide

File: README.md [hunk 4]
```diff
@@ -60,23 +57,17 @@ that node over IP address.
 3. All containers should be able to communicate with the k8s daemons running
 in the master node.
 
-  This is a tricky requirement when OVN is run in the "overlay" mode.  To
-  achieve this goal, we create a logical switch for the master node and also
-  create a logical port for the same node (represented by a OVS internal
-  interface and a private IP address).  We then use the logical port's IP
-  address as the IP address via which the nodes can reach k8s central daemons.
+  This can be achieved by running OVN gateways in the minion nodes. With
+  atleast one OVN gateway, the pods can reach the k8s central daemons with NAT.
```
Comment (tong101) [inline README.md]: s/atleast/at least

File: README.md [hunk 5]
```diff
@@ -86,301 +77,121 @@ database.
...
-And finally, start the ovn-controller.  (You need to run the below command on
-every boot)
+Also start ovn-controller on this node.
 
 ```
 /usr/share/openvswitch/scripts/ovn-ctl start_controller
 ```
 
-### k8s master node initialization.
+Now start the ovnkube utility on the master node too.
```
Comment (tong101) [inline README.md]: We start ovnkube on to init master first, do we still need "too" here?

File: README.md [hunk 6]
```diff
@@ -86,301 +77,121 @@ database.
...
-On the master node, with a unique node name of $NODE_NAME, for a cluster wide
-private address range of $CLUSTER_IP_SUBNET and a smaller subnet for the
-logical switch created in the master node of $MASTER_SWITCH_SUBNET, run:
-
-```
-ovn-k8s-overlay master-init \
-  --cluster-ip-subnet=$CLUSTER_IP_SUBNET \
-  --master-switch-subnet="$MASTER_SWITCH_SUBNET" \
-  --node-name="$NODE_NAME"
+ sudo ovnkube -k8s-kubeconfig kubeconfig.yaml -net-controller \
```
Comment (tong101) [inline README.md]: Is it worth providing a link to the sample kubeconfig.yaml here?

File: README.md [hunk 7]
```diff
@@ -86,301 +77,121 @@ database.
...
-git clone https://github.com/openvswitch/ovn-kubernetes
-cd ovn-kubernetes
-pip install .
-```
+On each host, you will need to run the following command once.
 
-You have two options with respect to choosing the interface via which
-North-South connectivity happens.
+The below command expects the user to provide
+* A cluster wide private address range of $CLUSTER_IP_SUBNET
```
Comment (tong101) [inline README.md]: Can $CLUSTER_IP_SUBNET on node be different with master? If not, is it worth mentioning $CLUSTER_IP_SUBNET should be the same as master?

Comment (rajatchopra) [reply README.md]: which brings the idea that maybe the master should store it and the node just learns it, through ovsdb or apiserver.

File: README.md (hunk 6 above)
Comment (shettyg) [reply README.md]: It is probably a overkill as people that start a kubernetes cluster know this.

**Comment by shettyg** [comment]
@tong101 
I took all your suggestions (except the sample kubeconfig.yaml). We can iteratively make the documentation better as we go along.

**Pull request merged**